low-level/files/opt/seagate/sspl/setup/sspl_config.py

```diff
@@ -0,0 +1,331 @@
+from cortx.utils.process import SimpleProcess
+from cortx.utils.validator.v_service import ServiceV
+from cortx.sspl.lowlevel.files.opt.seagate.sspl.setup.error import SetupError
+from cortx.utils.conf_store import Conf
+from cortx.sspl.bin.conf_based_sensors_enable import update_sensor_info
+
+
+class SSPLConfig:
+    """ SSPL Setup Config Interface """
+
+    name = "config"
+    DIR_NAME= "/opt/seagate/%s/sspl" % consts.PRODUCT_FAMILY
+    RSYSLOG_CONF ="/etc/rsyslog.d/0-iemfwd.conf"
+    RSYSLOG_SSPL_CONF = "/etc/rsyslog.d/1-ssplfwd.conf"
+    LOGROTATE_DIR  ="/etc/logrotate.d"
+    IEM_LOGROTATE_CONF = "%s/iem_messages" % LOGROTATE_DIR
+    SSPL_LOGROTATE_CONF = "%s/sspl_logs" % LOGROTATE_DIR
+    SSPL_CONFIGURED_DIR = "/var/%s/sspl" % consts.PRODUCT_FAMILY
+    SSPL_CONFIGURED = "%s/sspl-configured" % SSPL_CONFIGURED_DIR
+
+    def __init__(self, args : list):
+        self.args = args
+        self._script_dir = os.path.dirname(os.path.abspath(__file__))
+        self.role = None
+        Conf.load('sspl', 'yaml://%s' % consts.file_store_config_path)
+
+    def replace_expr(self, filename:str, key, new_str:str):
+        with open(filename, 'r+') as f: 
+            lines = f.readlines()
+            if type(key) == str:
+                for i, line in enumerate(lines):                 
+                    if re.search(key, line):
+                        lines[i] = re.sub(key, new_str, lines[i])
+            elif type(key) == int:
+                if not re.search(new_str, lines[key]):
+                    lines.insert(key, new_str)
+            else:
+                raise SetupError(
+                            errno.EINVAL,
+                            "Key data type : '%s', should be string or int",
+                            type(key))
+            f.seek(0)
+            for line in lines:
+                f.write(line)
+
+    def config_sspl(self):
+        if(os.geteuid() != 0):
+            raise SetupError(
+                        errno.EINVAL, 
+                        "Run this command with root privileges!!")
+        
+        if not os.path.isfile(consts.file_store_config_path):
+            raise SetupError(
+                        errno.EINVAL, 
+                        "Missing configuration!! Create and rerun.",
+                        consts.file_store_config_path)
+
+        # Put minion id, consul_host and consul_port in conf file
+        # Onward LDR_R2, salt will be abstracted out and it won't
+        # exist as a hard dependeny of SSPL
+        if consts.PRODUCT_NAME == "LDR_R1":
+            from cortx.sspl.bin.salt_util import SaltInterface
+            salt_util = SaltInterface()
+            salt_util.update_config_file(consts.file_store_config_path)
+
+        if os.path.isfile(self.SSPL_CONFIGURED):
+            os.remove(self.SSPL_CONFIGURED)
+
+        # Get Product
+        # product = self.getval_from_ssplconf('product')
+        product = Conf.get('sspl', 'SYSTEM_INFORMATION>product')
```

Comment:
Get `product` from global_config.yaml which is coming from args.

Refer: 
https://github.com/Seagate/cortx-monitor/pull/279/files#diff-e39efed6e2afb5011c788a62bb7f25d9c005e23bc3591fbd201fa42963897903R54